### PR TITLE
#38 issue  Add Docker Setup for Frontend and Backend

### PR DESCRIPTION
**Pull request by PankajKumardev**
#38  Issue

## 🔨 Changes Made
This PR introduces Docker configuration for both the frontend (React) and backend (Node.js) applications. The setup simplifies running the project in isolated containers for development.

Docker Setup Instructions:
Build containers: docker-compose build
Start containers: docker-compose up
Access frontend: http://localhost:3001 (updated for the new port)
Access backend: http://localhost:5000
Stop containers: docker-compose down

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings.
- [x] I have tested it locally and it works fine.
- [ ] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [x] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## 🤝 Related Issues

- Fixes #

## 📸 Screenshots (if applicable)

---

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to VigneshDevHub! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

**Comment by PankajKumardev** [comment]
@kartikayasija accept this pr and merge , and add labels

File: backend/Dockerfile [hunk 1]
```diff
@@ -3,10 +3,11 @@ FROM node:21
...
 COPY package.json yarn.lock ./
-RUN yarn
+
+RUN yarn install
 
 COPY . .
 
 EXPOSE 5000
 
-CMD ["node", "server.js"]
```
Comment (kartikayasijaa) [inline backend/Dockerfile:12]: you don't need to do this. don't change this file

File: frontend/dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM node:21
```
Comment (kartikayasijaa) [inline frontend/dockerfile:1]: is it working with node 21? Also we don't need a docker file for frontend. Don't add this in docker-compose

Comment (PankajKumardev) [reply frontend/dockerfile:1]: ok i  will revert back to backend and , compose to only setup mongo and backend.

**Comment by PankajKumardev** [comment]
@kartikayasijaa revert back to backend dockerfile and removed frontend dockerfile and made changes to compose.yml

**Comment by Sibam-Paul** [comment]
@PankajKumardev Please update ur branch

**Comment by PankajKumardev** [comment]
@Sibam-Paul done , i synced my branch

**Comment by PankajKumardev** [comment]
@kartikayasijaa  @Sibam-Paul  are you guys merging this or not?

**Comment by Sibam-Paul** [comment]
@PankajKumardev yes wait

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+version: '3.8'
+services:
+  backend:
+    build:
+      context: ./backend
+      dockerfile: Dockerfile
+    environment:
+      JWT_SECRET: "your_secret"
```
Comment (Sibam-Paul) [inline docker-compose.yml:8]: WILL THE SECRET be in .env file ??

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
Can u please make a pull request to test branch and please give a working ss or video and show me the ports working

**Comment by Sibam-Paul** [comment]
@PankajKumardev can't access the pork 3001 please make a working video

**Comment by Sibam-Paul** [comment]
@kartikayasijaa please review thir PR

**Comment by Sibam-Paul** [comment]
@PankajKumardev Please see discussion tab and join discord

**Comment by Sibam-Paul** [comment]
@PankajKumardev Only add mongodb in docker file nothing else and

**Comment by PankajKumardev** [comment]
@Sibam-Paul bro for mongodb u don't need docker file, if you create docker file  for backed and to integrate it with Mongo and for one command u can integrate them, I made that file already u have to just wrote docker compose up, it will start backend and Mongo

**Comment by PankajKumardev** [comment]
@Sibam-Paul  as you said about exposing of jwt token , now i fixed that also but make sure , env is made than only the server will run . and please do merge this PR

env example :
JWT_SECRET=your_secret_value_here
MONGO_URL=mongodb://mongo:27017/talk-trove
PORT=5000

**Comment by Sibam-Paul** [comment]
@PankajKumardev Just add MongoDB service with volume nothing else and put that in backend directory

**Pull request merged**